### ISSUE TITLE
JS filename not updated in script tag after hash change

### ISSUE THREAD

**Issue by mischnic** [parcel-bundler/parcel#3393]
`parcel index.html`:

```html
<script type="text/javascript" src="./index.js"></script>
```

```js
console.log("hey");
```

Changing the script file does make HMR reload the page, but the old script is still executed (also when doing a manual a manual refresh)
Restarting Parcel has no effect, deleting the cache updates the served script.

Running `parcel build index.html --no-scope-hoist --no-minify`, making a change and rerunning the build creates a new JS file (with a different hash and reflecting the changes) but doesn't update the JS filename in `index.html`.

2.0.0-alpha.1.1

**Comment by magcius** [comment]
Seeing this also. This unfortunately makes Parcel 2 kind of useless for development for me right now.

**Comment by mischnic** [comment]
Fixed by #3414.
